frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices-mode/install-wizard-steps/storage-and-nodes-step.tsx

```diff
@@ -81,34 +81,64 @@ export const StorageAndNodes: React.FC<StorageAndNodesProps> = ({ state, dispatc
     storageClass,
     storageClassName: scName,
     enableMinimal,
-    nodes,
     stretchClusterChecked,
     enableFlexibleScaling,
+    lvsSelectNodes = [],
+    lvsAllNodes = [],
   } = state;
 
-  const scNodeNames: React.MutableRefObject<string[]> = React.useRef([]); // names of the nodes, backing the storage of selected storage class
+  const memoizedPvData = useDeepCompareMemoize(pvData, true);
+  const memoizedNodesData = useDeepCompareMemoize(nodesData, true);
+  const selectedNodes = [...lvsSelectNodes, ...lvsAllNodes].map(getName);
```

Comment:
This will make all nodes as selected. For the state both `allNodes` and `selectNodes` are maintained separately.
If the user has select nodes then it does not means he will not have `allNodes`.
For dealing with nodes with no disks selected on previous step, suggest to use - `chartNodes`

Reply:
I think we should reset `allNodes` when `selectNodes` gets modified and vice versa.

Reply:
I dont want to have this refactor with this BZ, and as a preference we would have only one node state for lvs.